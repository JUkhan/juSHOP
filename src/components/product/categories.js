
import React, { useEffect } from 'react'
import { useSubscriptions } from '../../utils'
import { dispatch } from 'ajwah-store'
import { GetCategories, SelectCategory } from '../../store/actions'
import { Layout, Menu } from 'antd'

const { Sider } = Layout;

export function Categories() {
    const { category } = useSubscriptions(['category'])
    useEffect(() => { dispatch(GetCategories) }, [])

    return (
        <Sider style={{
            overflow: 'auto', height: '100vh', position: 'fixed', left: 0,
        }}
        >
            <div className="logo">juSHOP</div>
            <div className="categories">Categories</div>

            <Menu theme="dark" mode="inline" >
                {
                    category.data.map(item => <Menu.Item onClick={() => dispatch(SelectCategory, item)} key={item.category_id}>
                        <span className="nav-text">{item.name}</span>
                    </Menu.Item>)
                }
            </Menu>

        </Sider>
    )
}
