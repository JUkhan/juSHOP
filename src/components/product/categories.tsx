
import React, { useEffect } from 'react'
import { useSubscriptions } from 'react-ajwah'
import { dispatch } from 'ajwah-store'
import { GetCategories, SelectCategory } from '../../store/actions'
import { Layout, Menu } from 'antd'
import { ICategory, ICategoryState } from '../../store/model';

const { Sider } = Layout;

export function Categories() {
    const { category } = useSubscriptions<{ category: ICategoryState }>(['category'])
    useEffect(() => { dispatch(GetCategories) }, [])

    return (
        <Sider style={{
            overflow: 'auto', height: '100vh', position: 'fixed', left: 0,
        }}
        >
            <div className="logo">juSHOP</div>
            <div className="categories">Categories</div>

            <Menu theme="dark" mode="inline" >
                {
                    category.data.map((item: ICategory) => <Menu.Item onClick={() => dispatch(SelectCategory, item)} key={item.category_id}>
                        <span className="nav-text">{item.name}</span>
                    </Menu.Item>)
                }
            </Menu>

        </Sider>
    )
}
