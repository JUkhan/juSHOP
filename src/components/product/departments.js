
import React, { useEffect } from 'react'
import { useSubscriptions } from '../../utils'
import { dispatch } from 'ajwah-store'
import { GetDepartments, SelectDepartment } from '../../store/actions'
import { Menu } from 'antd'



export function Departments() {
    const { department } = useSubscriptions(['department'])
    useEffect(() => { dispatch(GetDepartments) }, [])

    return (
        <Menu theme="dark"
            mode="horizontal"
            style={{ lineHeight: '64px', textAlign: 'center' }} >
            {
                department.data.map(item => <Menu.Item onClick={() => dispatch(SelectDepartment, item)} key={item.department_id}>
                    <span className="nav-text">{item.name}</span>
                </Menu.Item>)
            }
        </Menu>


    )
}
