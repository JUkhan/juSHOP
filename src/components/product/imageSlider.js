
import React, { useState } from 'react'

export function ImageSlide({ image1, image2 }) {
    const [image, setImage] = useState(image1)
    function clickImage(img) {
        setImage(img)
    }
    return (
        <div style={{ textAlign: 'center' }}>
            <img width={200} alt="" src={'https://backendapi.turing.com/images/products/' + image} />
            <div style={{ borderBottom: '#ccc dotted thin', marginTop: 5, marginBottom: 5 }}></div>
            <img onClick={() => clickImage(image1)} style={{ paddingRight: 10, cursor: 'pointer' }} width={50} alt="" src={'https://backendapi.turing.com/images/products/' + image1} />
            <img onClick={() => clickImage(image2)} style={{ cursor: 'pointer' }} width={38} alt="" src={'https://backendapi.turing.com/images/products/' + image2} />
        </div>
    )
}