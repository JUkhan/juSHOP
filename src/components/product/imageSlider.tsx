
import React, { useState } from 'react'

export function ImageSlide({ image1, image2 }: any) {
    const [image, setImage] = useState(image1)
    function clickImage(img: any) {
        setImage(img)
    }
    return (
        <div style={{ textAlign: 'center' }}>
            <img className="img-content" width={200} alt="" src={'https://backendapi.turing.com/images/products/' + image} />
            <div style={{ borderBottom: '#ccc dotted thin', marginTop: 5, marginBottom: 5 }}></div>
            <img className="img1" onClick={() => clickImage(image1)} style={{ paddingRight: 10, cursor: 'pointer' }} width={50} alt="" src={'https://backendapi.turing.com/images/products/' + image1} />
            {image2 && <img className="img2" onClick={() => clickImage(image2)} style={{ cursor: 'pointer' }} width={38} alt="" src={'https://backendapi.turing.com/images/products/' + image2} />}
        </div>
    )
}