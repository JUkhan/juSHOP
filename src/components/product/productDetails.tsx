import React, { useEffect, useState } from 'react'
import { Api } from '../../services/Api';
import { take } from 'rxjs/operators';
import { Button, Row, Col } from 'antd'
import { AddToCart } from '../../store/actions';
import { dispatch } from 'ajwah-store'
import { updateObject } from '../../utils'
import { ImageSlide } from './imageSlider'
import { Review } from './review';

export function ProductDetails(props: any) {
    const [sa, setSAState] = useState<any>({})
    const [attributes, setAttributesState] = useState([])

    useEffect(() => {
        Api.getProductDetails(props.product.product_id).pipe(take(1)).subscribe(res => {
            props.product.description = res[0].description;
            props.product.image = res[0].image;
            props.product.image2 = res[0].image_2;
            setSAState({})

        });
        Api.getAttributes(props.product.product_id).pipe(take(1)).subscribe(res => setAttributesState(res))
    }, [])

    const colors = attributes.filter((item: any) => item.attribute_name === 'Color')
    const sizes = attributes.filter((item: any) => item.attribute_name === 'Size')
    const { name, description, price, discounted_price, thumbnail, image2, product_id } = props.product;

    function addToCart() {
        dispatch(AddToCart, props.product)
        props.hide();
    }
    function setColor(color: any) {
        setSAState(updateObject(sa, { color }))
        props.product.color = color
    }
    function setSize(size: any) {
        setSAState(updateObject(sa, { size }))
        props.product.size = size
    }

    return (
        <React.Fragment>
            <Row gutter={15}>
                <Col span={8}>
                    <ImageSlide image1={thumbnail} image2={image2} />
                </Col>
                <Col span={16}>
                    <h1>{name}</h1>
                    <div style={{ paddingBottom: 20 }}>
                        <span style={{ textDecoration: 'line-through' }}>{discounted_price === '0.00' ? null : '$' + price}</span>
                        <b style={{ fontSize: 20, marginLeft: 20, color: 'red' }}>{discounted_price === '0.00' ? '$' + price : '$' + discounted_price}</b>

                    </div>

                    <p>{description}</p>
                    <b>Color</b>
                    <div className="color-size">
                        <ul >
                            {colors.map((item: any) => <li key={name + item.attribute_value} ><Button className={sa.color === item.attribute_value ? 'circle-border' : ''} onClick={() => setColor(item.attribute_value)} shape="circle" style={{ backgroundColor: item.attribute_value }} /></li>)}
                        </ul>
                    </div>
                    <div />
                    <b>Size</b>
                    <div className="color-size">
                        <ul>
                            {sizes.map((item: any) => <li key={name + item.attribute_value} ><Button className={sa.size === item.attribute_value ? 'circle-border' : ''} onClick={() => setSize(item.attribute_value)} shape="round">{item.attribute_value}</Button></li>)}
                        </ul>
                    </div>
                    <div style={{ textAlign: 'center', marginTop: 20 }}>
                        <Button disabled={!(sa.size && sa.color)} onClick={addToCart} type="danger" shape="round">Add to Cart</Button>
                    </div>
                </Col>

            </Row>
            <Review productId={product_id} />

        </React.Fragment>
    )
}

