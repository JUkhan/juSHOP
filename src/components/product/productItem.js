import React, { useState } from 'react';
import { Card, Button, Modal } from 'antd';
import { ProductDetails } from './productDetails'


export function ProductItem(props) {
    const { product } = props;
    let [visible, setVisibleState] = useState(false)

    function hide() {
        setVisibleState(false)
    }
    function show() {
        setVisibleState(true)
    }
    return (
        <React.Fragment>
            <Card hoverable>
                <div style={{ textAlign: 'center' }}>
                    <img alt="example" src={'https://backendapi.turing.com/images/products/' + product.thumbnail} />
                    <h3>{product.name}</h3>
                </div>
                <div style={{ paddingTop: 15, paddingBottom: 15 }}>
                    <span style={{ textDecoration: 'line-through', float: 'left' }}>{product.discounted_price === '0.00' ? null : '$' + product.price}</span>
                    <Button onClick={show} type="danger" style={{ float: 'right' }}>{product.discounted_price === '0.00' ? '$' + product.price : '$' + product.discounted_price}</Button>
                </div>
                <div style={{ clear: 'both' }}></div>

                <p style={{ height: 50, fontSize: 11, paddingTop: 5, overflow: 'hidden' }}>{product.description.substring(0, 120) + '......'}</p>

            </Card >
            <Modal key={product.name} width={800} visible={visible} onCancel={hide} footer={null}>
                <ProductDetails hide={hide} product={product} />
            </Modal>
        </React.Fragment>
    );
}