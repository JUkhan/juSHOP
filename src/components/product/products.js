import React, { useEffect } from 'react'
import * as ActionNames from '../../store/actions'
import { dispatch } from 'ajwah-store'
import { List, Pagination } from 'antd'
import { ProductItem } from './productItem'
import { useSubscriptions } from 'react-ajwah';



export function Products() {

    const { product } = useSubscriptions(['product'])

    useEffect(() => {
        dispatch(ActionNames.LoadProducts, {})
    }, [])

    return (
        <React.Fragment>
            <Pagination style={{ marginBottom: 15, textAlign: 'center' }} showSizeChanger onChange={onPageChange} onShowSizeChange={onPageChange} current={product.pageNo} pageSize={product.limit} total={product.data.count} />
            <List
                itemLayout="horizontal"
                loading={product.message ? true : false}
                grid={{
                    gutter: 16, xs: 1, sm: 2, md: 4, lg: 4, xl: 4, xxl: 3,
                }}
                dataSource={product.data.rows}
                renderItem={item => (
                    <List.Item>
                        <ProductItem product={item}></ProductItem>
                    </List.Item>
                )}
            />
            <Pagination style={{ textAlign: 'center' }} showSizeChanger onChange={onPageChange} onShowSizeChange={onPageChange} current={product.pageNo} pageSize={product.limit} total={product.data.count} />
        </React.Fragment>
    );
}

function onPageChange(pageNo, limit) {
    dispatch(ActionNames.PageChange, { pageNo, limit })
}
