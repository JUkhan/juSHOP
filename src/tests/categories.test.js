import React from "react";
import { Categories } from '../components/product/categories'
import Enzyme, { mount } from 'enzyme'

import Adapte from 'enzyme-adapter-react-16'
import { storeCtx } from "ajwah-store";
import { take } from "rxjs/operators";


Enzyme.configure({ adapter: new Adapte() })


describe('categories', () => {

    test('categories - top menu should generate 7 li', async done => {
        const wrapper = mount(<Categories />)

        storeCtx().select('category').pipe(take(1)).subscribe(res => {
            expect(res.data.length).toBe(7)
            expect(wrapper.find('ul').text()).toBe(res.data.map(_ => _.name).join(''))
        }, done, done)

    })

})
