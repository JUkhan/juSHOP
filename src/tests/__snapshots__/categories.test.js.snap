// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`categories categories - top menu should generate 7 li 1`] = `
<aside
  className="ant-layout-sider ant-layout-sider-dark"
  style={
    Object {
      "flex": "0 0 200px",
      "height": "100vh",
      "left": 0,
      "maxWidth": "200px",
      "minWidth": "200px",
      "overflow": "auto",
      "position": "fixed",
      "width": "200px",
    }
  }
>
  <div
    className="ant-layout-sider-children"
  >
    <div
      className="logo"
    >
      juSHOP
    </div>
    <div
      className="categories"
    >
      Categories
    </div>
    <ul
      className="ant-menu ant-menu-dark ant-menu-root ant-menu-inline"
      onMouseEnter={[Function]}
      onTransitionEnd={[Function]}
      role="menu"
      style={Object {}}
    >
      <li
        className="ant-menu-item"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        role="menuitem"
        style={
          Object {
            "paddingLeft": 24,
          }
        }
      >
        <span
          className="nav-text"
        >
          French
        </span>
      </li>
      <li
        className="ant-menu-item"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        role="menuitem"
        style={
          Object {
            "paddingLeft": 24,
          }
        }
      >
        <span
          className="nav-text"
        >
          Italian
        </span>
      </li>
      <li
        className="ant-menu-item"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        role="menuitem"
        style={
          Object {
            "paddingLeft": 24,
          }
        }
      >
        <span
          className="nav-text"
        >
          Irish
        </span>
      </li>
      <li
        className="ant-menu-item"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        role="menuitem"
        style={
          Object {
            "paddingLeft": 24,
          }
        }
      >
        <span
          className="nav-text"
        >
          Animal
        </span>
      </li>
      <li
        className="ant-menu-item"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        role="menuitem"
        style={
          Object {
            "paddingLeft": 24,
          }
        }
      >
        <span
          className="nav-text"
        >
          Flower
        </span>
      </li>
      <li
        className="ant-menu-item"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        role="menuitem"
        style={
          Object {
            "paddingLeft": 24,
          }
        }
      >
        <span
          className="nav-text"
        >
          Christmas
        </span>
      </li>
      <li
        className="ant-menu-item"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        role="menuitem"
        style={
          Object {
            "paddingLeft": 24,
          }
        }
      >
        <span
          className="nav-text"
        >
          Valentine's
        </span>
      </li>
    </ul>
  </div>
</aside>
`;
